src/settings/toml/manifest.rs

```diff
@@ -85,60 +85,91 @@ impl Manifest {
                 Manifest::default()
-            }
-        };
+            });
 
-        let default_workers_dev = match &template_config.route {
-            Some(route) => {
-                if route.is_empty() {
-                    Some(true)
-                } else {
-                    None
-                }
+        config_template.warn_on_account_info();
+        if let Some(target_type) = &target_type {
+            if config_template.target_type != *target_type {
+                StdOut::warn(&format!("The template recommends the \"{}\" type. Using type \"{}\" may cause errors, we recommend changing the type field in wrangler.toml to \"{}\"", config_template.target_type, target_type, config_template.target_type));
             }
+        }
+
+        let default_workers_dev = match &config_template.route {
+            Some(route) if route.is_empty() => Some(true),
             None => Some(true),
+            _ => None,
         };
 
-        template_config.name = name;
-        template_config.workers_dev = default_workers_dev;
+        /*
+         * We use toml-edit for actually changing the template provided wrangler.toml,
+         * since toml-edit is a format-preserving parser. Elsewhere, we use only toml-rs,
+         * as only toml-rs supports serde.
+         */
+
+        let mut config_template_doc =
+            config_template_str
+                .parse::<toml_edit::Document>()
+                .map_err(|err| {
+                    failure::err_msg(format!(
+                        "toml_edit failed to parse config template. {}",
+                        err
+                    ))
+                })?;
+
+        config_template_doc["name"] = toml_edit::value(name);
+        if let Some(default_workers_dev) = default_workers_dev {
+            config_template_doc["workers_dev"] = toml_edit::value(default_workers_dev);
+        }
         if let Some(target_type) = &target_type {
-            template_config.target_type = target_type.clone();
+            config_template_doc["target_type"] = toml_edit::value(target_type.to_string());
         }
-
-        if let Some(arg_site) = site {
-            if template_config.site.is_none() {
-                template_config.site = Some(arg_site);
+        if let Some(site) = site {
+            if config_template.site.is_none() {
+                config_template_doc["site"]["bucket"] =
+                    toml_edit::value(site.bucket.to_string_lossy().as_ref());
+
+                if let Some(entry_point) = &site.entry_point {
+                    config_template_doc["target_type"] =
+                        toml_edit::value(entry_point.to_string_lossy().as_ref());
+                }
+                if let Some(include) = &site.include {
+                    let mut arr = toml_edit::Array::default();
+                    include.iter().for_each(|i| {
+                        arr.push(i.as_ref()).unwrap();
+                    });
```

Comment:
I think you can do this in one line:
```rust
let arr = toml_edit::Array::from(include.iter().map(String::as_ref).collect());
```
Same for `exclude` below.

Reply:
There's no `From<_>` impl that will make this work unfortunately.